pkg/pgmodel/model/samples.go

```diff
@@ -13,6 +13,13 @@ import (
 	"github.com/timescale/promscale/pkg/prompb"
 )
 
+// DataWithTS wraps incoming data with its in-timestamp. It is used to warn if the rate
+// of incoming samples vs outgoing samples is too low, based on time.
+type DataWithTS struct {
```

Comment:
We need to benchmark the memory impact of this (with an overview on GC activity)

Reply:
I have a feeling that we'll need at least an object pool